plugins/module_utils/network/sonic/argspec/snmp/snmp.py

```diff
@@ -0,0 +1,227 @@
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The arg spec for the sonic_snmp module
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+class SnmpArgs(object):  # pylint: disable=R0903
+    """The arg spec for the sonic_snmp module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {
+        'config': {
+            'options': {
+                'agentaddress': {
+                    'elements': 'dict',
+                    'options': {
+                        'interface': {'type': 'str'},
+                        'ip': {
+                            'required': True,
+                            'type': 'str'
+                        },
+                        'port': {'type': 'int'},
+                        'vrf': {'type': 'str'}
+                    },
+                    'type': 'list'
+                },
+                'community': {
+                    'elements': 'dict',
+                    'options': {
+                        'group': {'type': 'str'},
+                        'name': {
+                            'required': True,
+                            'type': 'str'
+                        }
+                    },
+                    'type': 'list'
+                },
+                'contact': {'type': 'str'},
+                'enable_trap': {
+                    'choices': [
+                        'auth-fail',
+                        'bgp',
+                        'config-change',
+                        'link-down',
+                        'link-up',
+                        'ospf',
+                        'all'
+                    ],
+                    'elements': 'str',
+                    'type': 'list'
+                },
+                'engine': {'type': 'str'},
+                'group': {
+                    'elements': 'dict',
+                    'options': {
+                        'access': {
+                            'elements': 'dict',
+                            'options': {
+                                'notify_view': {'type': 'str'},
+                                'read_view': {'type': 'str'},
+                                'security_level': {
+                                    'choices': [
+                                        'no-auth-no-priv',
+                                        'auth-no-priv',
+                                        'auth-priv'
+                                    ]
+                                },
+                                'security_model': {
+                                    'choices': [
+                                        'any',
+                                        'v2c',
+                                        'v3'
```

Comment:
Allowance needs to be added for the "usm" security model.

```suggestion
                                        'v3',
                                        'usm'
```